module/VuFind/tests/unit-tests/src/VuFindTest/ILS/Driver/AlmaTest.php

```diff
@@ -175,10 +175,20 @@ protected function createConnector(string $test, ?array $config = null, ?string
                 [
                     new \VuFind\Date\Converter(),
                 ]
-            )->onlyMethods(['makeRequest'])
+            )->onlyMethods(['makeRequest', 'debug'])
             ->getMock();
+        $config ??= $this->defaultDriverConfig;
         // Configure the stub
-        $this->driver->setConfig($config ?? $this->defaultDriverConfig);
+        $this->driver->setConfig($config);
+
+        // Add test for debugging function result, if enabled
+        foreach ($config['Debug']['log_function_result'] ?? [] as $function => $value) {
+            $this->driver->expects($this->once())->method('debug')->willReturnCallBack(
+                function ($msg, array $context = [], $prependClass = true) use ($function) {
+                    $this->assertEquals("$function result:", $msg);
```

Comment:
Should this be asserting the start of the string, not the whole string? Otherwise it seems to omit the actual result. I see that tests are passing on this branch, so maybe I'm misunderstanding something... but this doesn't quite look right to me.

Reply:
Ye, i simplified the test to just expect the debug function to be called with param <function> result: so it will throw error if it is not being called :)